File: JS/DiagonalDifference.test.js
/**
 * @version JAVASCRIPT ES6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { diagonalDifference } from './DiagonalDifference.js'

TEST('Starting DiagonalDifference test...', (t) => {
    //Test 1
    let want = 15
    let got = diagonalDifference( [[11, 2, 4], [4, 5, 6], [10, 8, -12]] )
    t.assert(want === got, "Expect: 15")
    //Test 2
    want = 2
    got = diagonalDifference( [[1, 2, 3], [4, 5, 6], [9, 8, 9]] )
    t.assert(want === got, "Expect: 2")

    t.end()
})

File: JAVA/DiagonalDifferenceTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;

import java.util.ArrayList;
import java.util.List;

import org.junit.Test;

public class DiagonalDifferenceTest {
    
    @Test
	public void testDiagonalDifference() throws Exception {
		//Test 1
		int want = 15;
        List<List<Integer>> ops = new ArrayList<>();
        List<Integer> v = new ArrayList<>();
        v.add(11);
        v.add(2);
        v.add(4);
        ops.add(v);
        v = new ArrayList<>();
        v.add(4);
        v.add(5);
        v.add(6);
        ops.add(v);
        v = new ArrayList<>();
        v.add(10);
        v.add(8);
        v.add(-12);
        ops.add(v);
		int got = DiagonalDifference.diagonalDifference( ops );
        assertEquals(want, got);
		//Test 2
        ops = new ArrayList<>();
        v = new ArrayList<>();
		want = 2;
        v.add(1);
        v.add(2);
        v.add(3);
        ops.add(v);
        v = new ArrayList<>();
        v.add(4);
        v.add(5);
        v.add(6);
        ops.add(v);
        v = new ArrayList<>();
        v.add(9);
        v.add(8);
        v.add(9);
        ops.add(v);
		got = DiagonalDifference.diagonalDifference( ops );
        assertEquals(want, got);
	}

}
